scripts/windows/setup/IotEdgeSecurityDaemon.ps1

```diff
@@ -123,163 +121,333 @@ function Install-SecurityDaemon {
         # Password used to access the container registry and pull the IoT Edge Agent image.
         [Parameter(ParameterSetName = 'Manual')]
         [Parameter(ParameterSetName = 'DPS')]
-        [SecureString] $Password,
-
-        # Splatted into every Invoke-WebRequest invocation. Can be used to set extra options.
-        #
-        # If -Proxy is also specified, it overrides the `-Proxy` key set in this hashtable, if any.
-        #
-        # Example:
-        #
-        #     Install-SecurityDaemon -InvokeWebRequestParameters @{ '-Proxy' = 'http://localhost:8888'; '-ProxyCredential' = (Get-Credential).GetNetworkCredential() }
-        [HashTable] $InvokeWebRequestParameters,
-
-        # Don't install the Moby CLI (docker.exe) to $MobyInstallDirectory. Only takes effect for -ContainerOs 'Windows'
-        [Switch] $SkipMobyCli,
-
-        # Local path to iotedged zip file. Only kept for backward compatibility. Prefer to set -OfflineInstallationPath instead.
-        [String] $ArchivePath
```

Comment:
You also need to restore this parameter, since QuickStart needs it.

Reply:
-SkipMobyCli or -ArchivePath, gibhub is not very helpful.

In case of -ArchivePath, would QuickStart pass a path to the CAB?

Reply:
`-ArchivePath`

AFAICT all the code paths and E2E tests pass in only the expanded directory, not the zip file. With your packaging change I assume this directory will now also contain `docker.exe` and `dockerd.exe`

@philipktlin Can you confirm that no code path in QuickStart and Windows E2E tests uses `Install-SecurityDaemon -ArchivePath` with a zip file?

Reply:
Philip agreed that there should be no code path that calls it with a zip file. So if you just treat it as an expanded directory and copy its contents to `$EdgeInstallDirectory` and `$MobyInstallDirectory` that would work. But I guess this will complicate the uninstall too...

Reply:
I am not familiar with the other scenarios, but I would strongly prefer to only support the CAB route. This PR has a script to create the CAB, if needed, manually.

Reply:
Actually, with your packaging change, it will be a path to a directory containing the CAB file.

Reply:
If you fix [the packages build,](https://msazure.visualstudio.com/b32aa71e-8ed2-41b2-9d77-5bc261222004/_build/results?buildId=20273574) we can try queueing a build + E2E test run with https://github.com/Azure/iotedge/commit/e87e57092f52771675c3518af987f54307f511ce and see if it works. If it works, then we don't need to add `-ArchivePath` back.

Reply:
In that case, ArchivePath could just be alias to the Offline parameter. The break is in the certificate permission, should be unblocked tomorrow morning. Otherwise I can revert to the older cert if you want to try today.

Reply:
>In that case, ArchivePath could just be alias to the Offline parameter.

Right. I was thinking of taking this opportunity to get rid of `-ArchivePath` entirely, especially since it is slightly backward-incompatible anyway since it no longer handles expanded directories. But keeping it as an alias is fine as well. I think it's unlikely that anyone was using it with expanded directories anyway - it only gets used in exceptional circumstances. Venkat can decide.

>The break is in the certificate permission

I need to run the test in a separate branch anyway, so I can just revert https://github.com/Azure/iotedge/pull/807/commits/0328fefe18a0dc7a713c081fdae4a003fdcca381 there. You don't have to revert it here.

Reply:
Yup, let's get rid of -ArchivePath since we have OfflineInstallPath now. Thanks!